### ISSUE TITLE
Request: Override RTL on a per-line basis for `RtlMoveVisually`

### ISSUE THREAD

**Issue by Billiam** [codemirror/CodeMirror#6666]
I have sections of an RTL document that I'd like to treat as LTR (ex: an rtl markdown document with ltr code blocks)

In a RTL document, I can currently assign a line class from my mode, and set the section to LTR with css. 

`RtlMoveVisually`, and other direction sensitive code, will still treat the whole document as RTL and cursor movement will be incorrect when moving between neutral and LTR characters, even without any RTL content in the line.

I think it would be a useful improvement to allow direction to be determined on a per-line basis.

My guess is that checks for `cm.doc.direction == 'rtl'` would be replaced with `getLineDirection(line)`, and delegate to a user method, falling back to the document direction. I'm not sure what the best approach is.

**Comment by marijnh** [comment]
An option that holds a function for the editor to use to determine a given line's direction might be a good idea. One potential complication there is dealing with outdated information—the function would at least be called again when the line changes, but how would we address a situation where the function depends on some data (say, a toggle widget in the interface) that changes? The editor won't know. One (not super elegant) option would be to require the code to call `cm.refresh()` at that point, I guess.

**Comment by Billiam** [comment]
Yeah, I think it makes sense for external data, like a toggle, to have to inform CM somehow.

I'm also experimenting with reversing left/right controls in keymaps, instead of using `RtlMoveVisually`, for RTL and mixed RTL documents, which (despite how hacky it is) feels more predictable.
